Source/MQTTnet/Implementations/MqttTcpChannel.cs

```diff
@@ -150,7 +156,7 @@ public async Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancellat
                 }
 
                 // Workaround for: https://github.com/dotnet/corefx/issues/24430
```

Comment:
Question: starting with .NET Core 3.0 this workaround isn't needed anymore, so the code could be conditionally compiled for a "legacy" target and for new targets.

W/o the workaround perf in general should be better, as the cost of creating, and disposing the cancellationtoken-registration can be avoided.

Reply:
If it is fixed we should add a compiler switch (NETCOREAPP3_0_OR_GREATER).